src/main/java/seedu/address/logic/commands/ExpenseCommand.java

```diff
@@ -130,7 +140,27 @@ public CommandResult execute(Model model) throws CommandException {
             model.addActivity(activity);
         }
 
-        return new CommandResult(String.format(MESSAGE_SUCCESS, persons.size(), successMessage.toString()));
+        return new CommandResult(String.format(MESSAGE_SUCCESS,
+                amount, payingPerson.getName(), description, successMessage.toString()));
+    }
+
+    /**
+     * Searches for a {@code Person} object from a given list of people using a search string.
+     * @param str The search string
+     * @param searchScope The {@code Person} list to search from
+     * @return The search result as a {@code Person} object
+     * @throws CommandException if the search result is not unique
+     */
+    public Person searchPerson(String str, List<Person> searchScope) throws CommandException {
```

Comment:
Is this method called outside the class? If not, it should be a private method

Reply:
idk i'm thinking of moving this method elsewhere at some point, cause currently the name searching is all over the place

Reply:
Is `searchScope` always the addressbook level currently? If yes, I created a method `findPersonByName` on modelmanager that lets you do an exact matching by a String. You can merge master to your branch and use that method to clean up your code a bit.

This way, I think you can keep the method as private and no need to extract out the logic into some other helper class. Addressbook/ModelManager provides the API to find users by keywords/exact match, and its up to the other class to decide how they want to use it.

Reply:
the issue is that the search scope is sometimes activity level, other times address book level so it gets kind of messy and i can't reuse directly

Reply:
Hmm that makes sense actually. But I think its ok to keep the method in there as a private method, I don't see any other logic in our app needing for such a specific search function (keyword/exact search in both AddressBook level and ActivityBook level).

Reply:
adjusted